lib/model/QubitPhysicalObject.php

```diff
@@ -326,4 +326,86 @@ public function getRelationsAsArray(int $typeId = null)
 
     return $results->fetchAll(PDO::FETCH_ASSOC);
   }
+
+  /**
+   * An quick update of an existing physical_object database row
+   *
+   * If passed data wouldn't alter the current db data, return false without
+   * writing to the database
+   *
+   * Currently only updates the following columns:
+   * - object.updated_at
+   * - physical_object.type_id
+   * - physical_object_i18n.location
+   *
+   * @param array $data a keyed array (column => value)
+   * @param PropelPDO $connection a connection object
+   */
+  public function quickUpdate(array $data, $connection = null)
+  {
+    $doUpdate = false;
+
+    $updateCols = [
+      'typeId',
+      'location',
+    ];
+
+    if (!isset($connection))
+    {
+      $connection = QubitTransactionFilter::getConnection();
+    }
+
+    foreach ($updateCols as $name)
+    {
+      if (!empty($data[$name]) && $data[$name] != $this->$name)
+      {
+        $doUpdate = true;
+        $newvals[$name] = $data[$name];
+      }
+      else
+      {
+        $newvals[$name] = $this->$name;
+      }
+    }
+
+    if (!$doUpdate)
```

Comment:
Does this indicate there was an error updating? If so maybe should be an exception? Dunno...

Reply:
No, not an error. If `$doUpdate` is false the incoming data matches the data already in the database, so there's no need to do an update.  I'll add a comment to the code to make this clearer.